leshan-server-core/src/main/java/org/eclipse/leshan/server/client/Client.java

```diff
@@ -171,33 +172,33 @@ public InetSocketAddress getRegistrationEndpointAddress() {
             return null;
         }
 
-        LinkObject[] res = Arrays.copyOf(objectLinks, objectLinks.length);
+        final LinkObject[] res = Arrays.copyOf(objectLinks, objectLinks.length);
 
```

Comment:
Please do not add final keyword everywhere.
In Leshan we flavour `final` for class attributes only.